flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -128,6 +128,12 @@
     /** The unique ID marking the specific execution instant of the task. */
     private final ExecutionAttemptID attemptId;
 
+    /**
+     * This field indicates whether the execution is the original execution of an execution vertex,
+     * i.e. it is created along with the creation of resetting of the execution vertex.
+     */
+    private final boolean original;
```

Comment:
Maybe we can add a `originalAttemptNumber` in `SpeculativeExecutionVertex` to record the original execution? And then we don't need to change the  logic in `Execution` and `ExecutionVertex`

Reply:
Good idea!